### ISSUE TITLE
Allow to configure proxy on a provider basis

### ISSUE THREAD

**Issue by FWest98** [rancher/terraform-provider-rancher2#982]
While it is possible to set the appropriate environment variables to get Go to use a proxy (as fixed in #57 ), it would be nice to explicitly support setting proxy settings in the provider configuration; since I only want the rancher connection to use the proxy and not my other connections.

Similar requests for another provider: https://github.com/hashicorp/terraform-provider-kubernetes/issues/1133. It is up to the individual providers to implement this: https://github.com/hashicorp/terraform/issues/17386#issuecomment-369800155.

**Comment by sboschman** [comment]
@a-blender , is getting a pr merged into 'norman' something you can assist with in any way? 🙏

**Comment by ericpromislow** [comment]
PR https://github.com/rancher/terraform-provider-rancher2/pull/1237 has been in draft since Sept. 2023

That PR is waiting on https://github.com/rancher/norman/pull/453 which has been double-approved and
looks ok, but I don't have authority to merge norman PRs

**Comment by MKlimuszka** [comment]
Putting this on hold until we decide which release has the QA capacity to cover it.

**Comment by damomurf** [comment]
@MKlimuszka https://github.com/rancher/norman/pull/453 has now been merged, is there likely to be any movement on this?

**Comment by matttrach** [comment]
This issue is a couple of years old now and can only reference unsupported versions, so I am going to close it, feel free to open a new issue if this is still an problem.
